main.go

```diff
@@ -0,0 +1,35 @@
+package storageconsul
```

Comment:
Because it not package `main` I would suggest to not name this file `main.go` 
Maybe `module.go` or `plugin.go` would be more appropriate?

Reply:
Bad habit from switching from different languages haha. module.go sounds good to me as Caddy calls them modules.